### PR TITLE
Enhanced Type Safety and Error Handling in DateHelpers

### PR DESCRIPTION
**Pull request by liby**
## Overview

The main objective of this PR is to enhance type safety and error handling in the DateHelpers module. It introduces a type assertion method to ensure that any `DateLike` input is either a Date object or a string, and it's not null or undefined.

## Pull Request Type

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Current Behavior

Previously, there was less stringent checking of input types and no specific method to ensure that inputs to the DateHelpers functions were valid and of expected types.

## Changes Introduced

- **ensureValidDateInput**: A new private static method added in `DateHelpers` class. It performs type assertions to make sure the input is either a Date object or a string. It also checks if the input is not null or undefined. 
- **Jest Configuration**: Jest configuration is updated. Replaced `ts-jest` with `babel` for better compatibility and performance.
- **tsconfig Updates**: Introduced two separate tsconfig settings. `tsconfig.json` is for development purposes and `tsconfig.build.json` is used for building the package.
- **Tests for DateHelpers**: Comprehensive tests have been added for `date-helpers`. It ensures total code coverage. Removed Sentry capturing in the codebase in favor of direct error throwing.

## Commits Overview

- 695cb99cb6d2622965419dac9d73afd814bc8455: Enhanced input validation in `DateHelpers` with `ensureValidDateInput`. Incorporated it in both `parseDateString` and `formatDateLikeToString` methods to ensure input validity right at the start.
- 4c8c247b2093cd768eb702ae6f4f7d73a7f4fa21: Updated Jest configuration to use `babel` over `ts-jest`.
- 3260bda21591c0733d8405f4cee8536122766f29: Updated `tsconfig.json` configurations for better compatibility with mono repo structure. Introduced separate configurations for development (`tsconfig.json`) and for building (`tsconfig.build.json`).
- 5d88e0f83aca488dd96880a0b06b07437e886963: Added tests for `date-helpers` ensuring full code coverage. Also, removed Sentry capturing and replaced with direct error throws.

## Breaking Changes

- [ ] Yes
- [x] No

## Additional Notes

None.

### CODE REVIEW

**Comment by rainx** [comment]
Can you provide some tests for this exception condition?

File: packages/date-helpers/__tests__/date-helpers.test.ts [hunk 1]
```diff
@@ -1,9 +1,65 @@
...
+  it('parseDateString should throw for unrecognized date formats', () => {
+    const invalidDateString = 'ABCDE';
+    expect(() => DateHelpers.parseDateString(invalidDateString)).toThrowError(
+      InvalidArgumentException,
+    );
+  });
+
+  it('parseDateString should throw error for null or undefined input', () => {
+    // @ts-expect-error - intentionally passing invalid input to test error handling
+    expect(() => DateHelpers.parseDateString(null)).toThrow(
```
Comment (lixiaoyan) [inline packages/date-helpers/__tests__/date-helpers.test.ts:17]:
```suggestion
    expect(() => DateHelpers.parseDateString(null as unknown as DateLike)).toThrow(
```

File: tsconfig.json [hunk 2]
```diff
@@ -13,5 +13,5 @@
     "sourceMap": true,
     "skipLibCheck": true
   },
-  "exclude": ["node_modules", "**/*.spec.ts"]
+  "exclude": ["node_modules"]
```
Comment (lixiaoyan) [inline tsconfig.json:16]: nit: `node_modules` is excluded by default.
https://www.typescriptlang.org/tsconfig#exclude

File: packages/exceptions/src/exceptions/base.exception.ts
Comment (lixiaoyan) [inline packages/exceptions/src/exceptions/base.exception.ts:1]: nit: Should the constructor's `cause` parameter also have a `unknown` type?

File: packages/date-helpers/__tests__/date-helpers.test.ts [hunk 3]
```diff
@@ -35,12 +91,22 @@ describe('DateHelpers', () => {
...
+  it('formatDateLikeToUsLocaleMediumDateString should format dates correctly', () => {
     expect(DateHelpers.formatDateLikeToUsLocaleMediumDateString(output)).toBe(
       'Oct 21, 2019',
     );
     expect(
       DateHelpers.formatDateLikeToUsLocaleMediumDateString('2019-10-21'),
     ).toBe('Oct 21, 2019');
   });
+
+  it('formatDateLikeToString should return an empty string if the input is an empty string', () => {
```
Comment (lixiaoyan) [inline packages/date-helpers/__tests__/date-helpers.test.ts:103]: It's strange that it doesn't throw in this case though.

File: packages/exceptions/src/exceptions/base.exception.ts
Comment (rainx) [reply packages/exceptions/src/exceptions/base.exception.ts:1]: 已经在 https://github.com/RightCapitalHQ/frontend-libraries/pull/27 里处理

File: packages/date-helpers/__tests__/date-helpers.test.ts (hunk 1 above)
Comment (liby) [reply packages/date-helpers/__tests__/date-helpers.test.ts:17]: `as unknown as string`? I believe @ts-expect-error is clearer for intentional test errors.🤔

Comment (frantic1048) [reply packages/date-helpers/__tests__/date-helpers.test.ts:17]: In this particular scenario, we are indeed passing a `null` value, which does not conform to the expected `DateLike` type. Therefore, it appears that using `@ts-expect-error` would be more appropriate. 🤔

File: packages/date-helpers/__tests__/date-helpers.test.ts (hunk 3 above)
Comment (liby) [reply packages/date-helpers/__tests__/date-helpers.test.ts:103]: If we no longer accept empty strings here, it should be a breaking change. It should be handled in a separate PR.

**Review by lixiaoyan — APPROVED** [review]

**Pull request merged**